e2e/nomostest/portforwarder/port_forwarder.go

```diff
@@ -233,14 +223,29 @@ func (pf *PortForwarder) start() error {
 	// Restart the port forward if the process exits prematurely (e.g. due to pod eviction)
 	go func() {
 		for {
-			if err := pf.watcher.WatchForNotFound(kinds.Pod(), pf.pod, pf.ns, testwatcher.WatchTimeout(0), testwatcher.WatchContext(pf.ctx)); err != nil {
-				if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
-					// stop retrying, the subprocess will be killed with the context.
-					return
-				}
-				// Watch automatically retries, so this shouldn't happen unless
-				// there's a terminal error.
-				pf.logger.Infof("Watch exited, restarting port-forward for %s/%s: %v", pf.ns, pf.deployment, err)
+			// create an inner context for the watch so we can cancel threads between iterations
+			// the subprocess will be killed inside portForwardToDeployment
+			watchCtx, watchCancel := context.WithCancel(pf.ctx)
```

Comment:
always need a `defer cancel()` immediately after creating one.

Reply:
I'd need to put this in another nested function to use `defer`, but yeah that might be cleaner

Reply:
done